conformance/lang/values-types-and-variables/array.balt

```diff
@@ -0,0 +1,99 @@
+Test-case: output
+Description: Test valid array types and values
+Label: array
+
+const LENGTH = 2;
+
+public function main() returns () {
+    int[5] a = [1, 2, 3, 4, 5];
+    io:println(a); // @output [1,2,3,4,5]
+
+    byte[LENGTH] b = [1, 255];
+    io:println(b); // @output [1,255]
+
+    string[*] c = ["Cat", "Dog"];
+    io:println(c); // @output ["Cat","Dog"]
+
+    float[] d = [1.25, 2.5];
+    io:println(d); // @output [1.25,2.5]
+
+    d = [1.5];
+    io:println(d); // @output [1.5]
+
+    record {|string name;|}[2] e = [{name: "John"}, {name: "Mike"}];
+    map<string>[2] f = e;
+    io:println(f); // @output [{"name":"John"},{"name":"Mike"}]
+
+    any[6] g = [2, 2.3, "String", true, [], {}];
+    io:println(g); // @output [2,2.3,"String",true,[],{}]
+
+    ()[*] h = [null, ()];
+    io:println(h); // @output [(),()]
+
+    xml[LENGTH] i = [xml ``, xml `<body>v</body>`];
+    io:println(i); // @output [xml``,xml`<body>v</body>`]
+
+    int[4] j = [55555, 5343434, 999999, 4];
+    j[3] = 888888888;
+    io:println(j); // @output [55555,5343434,999999,888888888]
+
+    object {int id;}[] k = [object {int id = 3;}];
+    io:println(k[0].id); // @output 3
+
+    table<record {|int id;|}>[] l = [table [{id: 1001}, {id:1002}]];
+    io:println(l); // @output [table key() [{"id":1001},{"id":1002}]]
+
+    (int[])[2] m = [[45], [444, 55, 545]];
+    io:println(m); // @output [[45],[444,55,545]]
+
+    int[2] n = [1, 2];
+    int[] o = n;
+    o[1] = 433;
+    io:println(n); // @output [1,433]
+}
+
+Test-case: output
+Description: Initialization of array of array gives compilation error
```

Comment:
Shall we include in the description what we are testing (e.g., multidimensional array or array with an array member type) assuming there is no bug, before or as opposed to describing the current behaviour?